### PR TITLE
Export coalesce

### LINKED ISSUE #71: coalesce isn't exported
I'm not sure if there was a conscious decision behind this or not, but `coalesce` is not exported. I had the hardest time figuring out why we didn't have this functionality in Julia already until I was told that the function most definitely lives in this package. Doing a REPL help search for `Missings.coalesce` indeed shows that it does. I thought I'd just bring this to your attention if it hadn't already been mentioned.

### PR DESCRIPTION
**Pull request by ararslan**
Fixes #71. Not sure why this wasn't exported before.

### CODE REVIEW

File: src/Missings.jl [hunk 1]
```diff
@@ -30,7 +30,7 @@ end
 import Base: ==, !=, <, *, <=, !, +, -, ^, /, &, |, xor
 
 if VERSION >= v"0.7.0-DEV.2762"
-    using Base: ismissing, missing, Missing, MissingException
+    using Base: coalesce, ismissing, missing, Missing, MissingException
```
Comment (nalimilan) [inline src/Missings.jl]: IIRC `coalesce` has been added later, when I removed `Nullable`, so it should be under a different version check.

**Comment by quinnj** [comment]
I had a fix for this locally along w/ some other 0.7 compat. Any objections to merging?

File: src/Missings.jl [hunk 2]
```diff
@@ -441,29 +471,7 @@ julia> coalesce.([missing, 1, missing], [0, 10, 5])
...
-    levs = convert(AbstractArray{T}, filter!(!ismissing, unique(x)))
-    if method_exists(isless, Tuple{T, T})
-        try; sort!(levs); end
-    end
-    levs
-end
-
-# Deprecations
-@deprecate skip(itr) skipmissing(itr) false
+Compat.coalesce(x::Missing) = missing
```
Comment (nalimilan) [inline src/Missings.jl]: This should only be defined before Base includes `coalesce`, or it's going to overwrite the equivalent Base method.

Comment (quinnj) [reply src/Missings.jl]: That assumes that Compat already defines `coalesce(x::Missing)`, which it currently doesn't (it's commented out). It really boils down to whether the actual compat definition is in Missings or Compat.

Comment (nalimilan) [reply src/Missings.jl]: Right, but I was speaking about a different issue, which is that recent Julia 0.7 versions define that method, and that `Compat.coalesce = Base.coalesce` there.

Comment (quinnj) [reply src/Missings.jl]: I don't think so? I don't see any overwrite happening on latest master?

Comment (ararslan) [reply src/Missings.jl]: Easy enough to just wrap this in `is !isdefined(Base, :coalesce)` regardless.

**Pull request merged**